kernel/page_allocator/src/lib.rs

```diff
@@ -139,6 +121,222 @@ pub fn init(end_vaddr_of_low_designated_region: VirtualAddress) -> Result<(), &'
 	Ok(())
 }
 
+/// An enum used to wrap the generic PageRange variants corresponding to different page sizes.
+/// Additional methods corresponding to PageRange methods are provided in order to destructure the enum variants.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PageRangeSized {
+	Normal4KiB(PageRange),
+	Huge2MiB(PageRange<Page2MiB>),
+	Huge1GiB(PageRange<Page1GiB>),
+}
+
+// These methods mostly destructure the enum in order to call internal methods
```

Comment:
The range() and range_*() are present on `PageRangeSized` because the enum needs to be destructured before you can call methods on the internal `PageRange,` which could be one of several types. I didn't really see a way to get around adding these extra methods for this reason. Although dynamic dispatch could be one possible alternative, I figured that wasn't appropriate here.


`into_range_4k()` is for `Drop for AllocatedPages,` where a huge `PageRange` will be converted to a 4kb `PageRange` before anything. This is so that the allocator doesn't need any new logic for dropping huge pages.
I suppose that you could get the `PageRange from `PageRangeSized` and then convert to a 4kb page range, but that means you would have to check page size and call the correct range method, then convert to a 4kb range and make a new `PageRangeSized`,  whereas with `into_range_4k()` you just convert directly if the `PageRange` is huge. Either method works though, and I don't know if either would be better.